File: frontend/src/service/ocp-express/ObClusterController.ts
/*
 * Copyright (c) 2023 OceanBase
 * OCP Express is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */

/* eslint-disable */
// 该文件由 OneAPI 自动生成，请勿手动修改！
import request from '@/util/request';

/** 此处后端没有提供注释 GET /api/v1/ob/cluster */
export async function getClusterInfo(options?: { [key: string]: any }) {
  return request<API.SuccessResponse_ClusterInfo_>('/api/v1/ob/cluster', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/ob/cluster/unitSpecLimit */
export async function getClusterUnitSpecLimit(options?: { [key: string]: any }) {
  return request<API.SuccessResponse_ClusterUnitSpecLimit_>('/api/v1/ob/cluster/unitSpecLimit', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/ob/cluster/server */
export async function getServerInfo(
  params: {
    // query
    ip?: string;
    obSvrPort?: number;
  },
  options?: { [key: string]: any },
) {
  return request<API.SuccessResponse_Server_>('/api/v1/ob/cluster/server', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/v1/ob/cluster/init */
export async function initObCluster(body?: API.ClusterInitParam, options?: { [key: string]: any }) {
  return request<API.SuccessResponse_BasicCluster_>('/api/v1/ob/cluster/init', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/ob/cluster/charsets */
export async function listCharsets(
  params: {
    // query
    /** Tenant compatible mode. */
    tenantMode?: API.TenantMode;
  },
  options?: { [key: string]: any },
) {
  return request<API.IterableResponse_Charset_>('/api/v1/ob/cluster/charsets', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

File: server/src/main/java/com/oceanbase/ocp/controller/ObClusterController.java
/*
 * Copyright (c) 2023 OceanBase
 * OCP Express is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */

package com.oceanbase.ocp.controller;

import javax.validation.Valid;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.oceanbase.ocp.core.ob.cluster.BasicCluster;
import com.oceanbase.ocp.core.ob.tenant.TenantMode;
import com.oceanbase.ocp.core.response.IterableResponse;
import com.oceanbase.ocp.core.response.ResponseBuilder;
import com.oceanbase.ocp.core.response.SuccessResponse;
import com.oceanbase.ocp.obops.cluster.ClusterCharsetService;
import com.oceanbase.ocp.obops.cluster.ClusterInitService;
import com.oceanbase.ocp.obops.cluster.ClusterService;
import com.oceanbase.ocp.obops.cluster.ClusterUnitService;
import com.oceanbase.ocp.obops.cluster.model.Charset;
import com.oceanbase.ocp.obops.cluster.model.ClusterInfo;
import com.oceanbase.ocp.obops.cluster.model.ClusterUnitSpecLimit;
import com.oceanbase.ocp.obops.cluster.model.Server;
import com.oceanbase.ocp.obops.cluster.param.ClusterInitParam;

import lombok.extern.slf4j.Slf4j;

@Validated
@Slf4j
@RestController
@RequestMapping("/api/v1")
public class ObClusterController {

    @Autowired
    private ClusterInitService clusterInitService;

    @Autowired
    private ClusterService clusterService;

    @Autowired
    private ClusterCharsetService clusterCharsetService;

    @Autowired
    private ClusterUnitService clusterUnitService;

    @PostMapping(value = "/ob/cluster/init", produces = {"application/json"}, consumes = {"application/json"})
    public SuccessResponse<BasicCluster> initObCluster(@Valid @RequestBody ClusterInitParam param) {
        return ResponseBuilder.single(clusterInitService.initCluster(param));
    }

    @GetMapping(value = "/ob/cluster", produces = {"application/json"})
    public SuccessResponse<ClusterInfo> getClusterInfo() {
        return ResponseBuilder.single(clusterService.getClusterInfo());
    }

    @GetMapping(value = "/ob/cluster/server", produces = {"application/json"})
    public SuccessResponse<Server> getServerInfo(@RequestParam("ip") String ip,
            @RequestParam("obSvrPort") Integer obSvrPort) {
        return ResponseBuilder.single(clusterService.getServerInfo(ip, obSvrPort));
    }

    @GetMapping(value = "/ob/cluster/charsets", produces = {"application/json"})
    public IterableResponse<Charset> listCharsets(
            @Valid @RequestParam(value = "tenantMode", required = false) TenantMode tenantMode) {
        return ResponseBuilder.iterable(clusterCharsetService.listCharsets(tenantMode));
    }

    @GetMapping(value = "/ob/cluster/unitSpecLimit", produces = {"application/json"})
    public SuccessResponse<ClusterUnitSpecLimit> getClusterUnitSpecLimit() {
        return ResponseBuilder.single(clusterUnitService.getClusterUnitSpecLimit());
    }

}
